yjit_codegen.c

```diff
@@ -4449,6 +4449,31 @@ gen_getclassvariable(jitstate_t* jit, ctx_t* ctx, codeblock_t* cb)
     return YJIT_KEEP_COMPILING;
 }
 
+VALUE
+rb_vm_setclassvariable(const rb_iseq_t *iseq, const rb_cref_t *cref, const rb_control_frame_t *cfp, ID id, VALUE val, ICVARC ic);
+
+static codegen_status_t
+gen_setclassvariable(jitstate_t* jit, ctx_t* ctx, codeblock_t* cb)
+{
+    // rb_vm_setclassvariable can raise exceptions.
+    jit_prepare_routine_call(jit, ctx, REG0);
+
+    mov(cb, C_ARG_REGS[0], member_opnd(REG_CFP, rb_control_frame_t, ep));
+    call_ptr(cb, REG0, (void *)rb_vm_get_cref);
```

Comment:
Could use a comment here explaining what the cref call is for. What does it do?

Reply:
After looking at this we realized that we can refactor get/set class variable to make adding the cref here unnecessary. After https://github.com/ruby/ruby/pull/5137 is merged I'll refactor setclassvariable and update this PR.

Reply:
Wonderful. Thank you Eileen! :)

Reply:
Ok with https://github.com/ruby/ruby/pull/5137 and https://github.com/ruby/ruby/pull/5143 merged, I've updated this PR to remove `rb_vm_get_cref`.